### DISCUSSION TITLE
Sonoff SNZB-01 Setup [resolved by 1.22.0]

### DISCUSSION THREAD

**Discussion by floatiepen** [Koenkk/zigbee2mqtt#9336] [Devices]
I’m able to pair my three SNZB-01’s in zigbee2mqtt, but they don’t seem to do anything when pressed (with or without an automation configured). I tried watching MQTT ("#") and also opening the HA Developer pane and listening for events ("*"), but I don’t see anything from the buttons. Is there anything else I need to do to set them up?

I had them configured under ZHA, but then after I moved everything to z2m, ~30 other devices are fine, just these three buttons aren't working.  I can see them in the web interface, and rename them.  I've removed and re-added a few times.  the basic remove option doesn't seem to work, I need to use "Force remove".  I get this error when I try the basic (non-force) remove:
```
Zigbee2MQTT:error 2021-10-25 15:41:42: Request 'zigbee2mqtt/bridge/request/device/remove' failed with error: 'Failed to remove device 'Button 2 Office' (block: false, force: false) (Error: AREQ - ZDO - mgmtLeaveRsp after 10000ms)'
```

I deleted and re-added again.  this time, watching the z2m console.  No errors reported in the web UI, but the console shows this message:
```
Zigbee2MQTT:error 2021-10-25 15:41:52: Failed to configure 'Button 2 Office', attempt 3 (Error: Bind 0x00124b0022ea4060/1 genOnOff from '0x00124b001cde525b/1' failed (AREQ - ZDO - bindRsp after 10000ms)
    at Timeout._onTimeout (/app/node_modules/zigbee-herdsman/src/utils/waitress.ts:64:35)
    at listOnTimeout (internal/timers.js:557:17)
    at processTimers (internal/timers.js:500:7))
```

Is there some force-reset I can try?

**Comment by paperdigits** [comment]
I am having the same problem, the button appears to pair but then I don't see any events from it..

**Comment by c0mplex1** [comment]
I had also many problems to pair and use some of my SNZB-01 which ended in throwing the devices away.
I use now a TuYa TS0041 switches which are more reliable then the SONOFF junk.

**Comment by floatiepen** [comment]
I know the buttons work well because I used them for months with ZHA.  After switching to z2m, they still pair (and re-pair) just fine.  I'm just not seeing any event trigger.  I'm hoping it's something simple I'm missing, but I don't know where to look.

**Comment by paperdigits** [comment]
I'm already using several of these buttons which were paired previously to z2m, just new ones don't seem to work.

**Comment by jvo-cat** [comment]
After the change from Zigbee2Mqtt to 2021.21.0 I began to struggle to do automations with "button" devices that use the "action" trigger (Sonoff SNZB-01, Tuya Scene Switch 4 buttons and Aqara Magic Cube) ... I received recently, but I have almost a year with a Sonoff SNZB-01 which to date still works well for me ... The problem is that when I want to do an "Automation" and put the new ones as "Device" and want to assign it a "Trigger", it only shows the Battery and Signal options, and no longer shows, as in the previous one, that the "action" ... "single" ... "double" and "hold" options still work. I can't find in which part of HA I would have to "enable those entities" that previously in the automation code came as:
"trigger:
  - platform: device
    domain: mqtt
    device_id: XXXXX
    type: action
    subtype: single
    discovery_id: XXXXX action_single "

... and / or if the way to use these sensors already paired with version 2021.21.0, in automations is no longer the same ... And if so, what to do in these cases to be able to use them in automations that is their use logical ... I suppose there are going to be a lot of HA Zigbee2Mqtt users who come across this ... As always, your time and attention .. Thanks and regards !!

**Comment by paperdigits** [comment]
This is fixsed for me as of 2021.22.0 which came out today. I successfully paired two new buttons, which brings the total working number to 4.

**Comment by floatiepen** [comment]
Me too.  I just updated my docker container (1.22.0 commit: 0a1fccf) and removed and re-paired all three of my buttons.  Now they all show events triggered in the log, and all my home assistant automations using [AramidX's blueprint](https://github.com/AramidX/ha-blueprint/blob/dc4b846bc4dfa8725526606b07c051947bfc9687/blueprints/sonoff/sonoff_snzb01_wireless_switch_z2m.yaml) just started working!  Thanks for the update!

**Reply by jvo-cat** [reply]
Hello ... When doing an automation with them they appear in the Trigger ... "single" action ... "double" action and "hold" action?

**Reply by jvo-cat** [reply]
Excellent recommendation !! .. Use this Blueprint and I could do the automation ... Thank you !!!

**Reply by floatiepen** [reply]
I'm glad you got it working.  my old setup needed to reference the IEEE address, so this blueprint felt a lot cleaner to me too.

**Reply by upandabout** [reply]
How did you use that blue print? I'm having exactly the same problems with that switch...
